src/main/UI.ts

```diff
@@ -0,0 +1,190 @@
+'use strict';
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *  https://github.com/microsoft/vscode-extension-samples/
+ *--------------------------------------------------------------------------------------------*/
+
+import { QuickPickItem, window, Disposable, CancellationToken, QuickInputButton, QuickInput, ExtensionContext, QuickInputButtons, Uri } from 'vscode';
+
+
+// -------------------------------------------------------
+// Helper code that wraps the API for the multi-step case.
+// -------------------------------------------------------
+class InputFlowAction {
+	private constructor() { }
+	static back = new InputFlowAction();
+	static cancel = new InputFlowAction();
+	static resume = new InputFlowAction();
+}
+
+export class MyButton implements QuickInputButton {
+    constructor(public iconPath: { light: Uri; dark: Uri; }, public tooltip: string) { }
+}
+
+type InputStep = (input: MultiStepInput) => Thenable<InputStep | void>;
+
+interface QuickPickParameters<T extends QuickPickItem> {
+	title: string;
+	step: number;
+	totalSteps: number;
+	items: T[];
+	activeItem?: T;
+	placeholder: string;
+	buttons?: QuickInputButton[];
+	shouldResume: () => Thenable<boolean>;
+}
+
+interface InputBoxParameters {
+	title: string;
+	step: number;
+	totalSteps: number;
+	value: string;
+	prompt: string;
+	validate: (value: string) => Promise<string | undefined>;
+	buttons?: QuickInputButton[];
+	shouldResume: () => Thenable<boolean>;
+}
+
+export class MultiStepInput {
+
+	static async run<T>(start: InputStep) {
+		const input = new MultiStepInput();
+		return input.stepThrough(start);
+	}
+
+	private current?: QuickInput;
+	private steps: InputStep[] = [];
+
+	private async stepThrough<T>(start: InputStep) {
+		let step: InputStep | void = start;
+		while (step) {
+			this.steps.push(step);
+			if (this.current) {
+				this.current.enabled = false;
+				this.current.busy = true;
+			}
+			try {
+				step = await step(this);
+			} catch (err) {
+				if (err === InputFlowAction.back) {
+					this.steps.pop();
+					step = this.steps.pop();
+				} else if (err === InputFlowAction.resume) {
+					step = this.steps.pop();
+				} else if (err === InputFlowAction.cancel) {
+					step = undefined;
+				} else {
+					throw err;
+				}
+			}
+		}
+		if (this.current) {
+			this.current.dispose();
+		}
+	}
+
+	public async showQuickPick<T extends QuickPickItem, P extends QuickPickParameters<T>>({ title, step, totalSteps, items, activeItem, placeholder, buttons, shouldResume }: P) {
+		const disposables: Disposable[] = [];
+		try {
+			return await new Promise<T | (P extends { buttons: (infer I)[] } ? I : never)>((resolve, reject) => {
```

Comment:
I think you can remove the `async` and `await` keywords here, they don't seem to be doing anything

Reply:
Cross verified, after removing these keywords, UI panel does not perform anymore.